example/phylo/phylo2.jl

```diff
@@ -173,30 +174,63 @@ function bwfiltertree!(Q, tree::Tree, θlin, dt0)
 end
 
 
+"""
+    pCN(noise::AbstractNoiseProcess, ρ; reset=true,reverse=false,indx=nothing)
+Create a new, but correlated noise process from `noise` and additional entropy with correlation ρ.
+"""
+function DiffEqNoiseProcess.pCN(source::DiffEqNoiseProcess.AbstractNoiseProcess{T,N,Vector{T2},inplace}, ρ;
+              reset=true,reverse=false,indx=nothing) where {T,N,T2,inplace}
+
+  source′ = deepcopy(source)
+  rng = Random.GLOBAL_RNG
+  # generate new Wiener process similar to the one in source
+  t = source′.t
+  if typeof(source′.W[1]) <: Number
+    Wnew = cumsum([zero(source′.W[1]);[sqrt.(t[i+1]-ti).*DiffEqNoiseProcess.wiener_randn(rng,typeof(source′.W[i]))
+              for (i,ti) in enumerate(t[1:end-1])]])
+  else
+    Wnew = cumsum([[zero(source′.W[1])];[sqrt.(t[i+1]-ti).*DiffEqNoiseProcess.wiener_randn(rng,source′.W[i])
+              for (i,ti) in enumerate(t[1:end-1])]])
+  end
+
+  source′.W = ρ * source′.W + sqrt(one(ρ)-ρ^2) * Wnew
+  source′.u = ρ * source′.u + sqrt(one(ρ)-ρ^2) * Wnew
+  #NoiseGrid(t, Vector.(W))
+  NoiseWrapper(source′,reset=reset,reverse=reverse,indx=indx)
+end
 
 ## simulate conditioned process using messages
 
-function fwguidtree!(messages, X, guidedsegs, ll, tree::Tree, f, g, θ, rho)
+function fwguidtree!(messages, X, guidedsegs, noisesegs, ll, tree::Tree, f, g, θ, rho)
     T = tree.T
+    ll .= -1
+    ll[1] = 0.0
+    if isnothing(noisesegs)
+        noisesegs = guidedsegs
+    end
     for i in eachindex(T)
         i == 1 && continue  # skip root-node (has no parent)
         i′ = tree.Par[i]
+        @assert i′ < i
+    #    @show i′, i
+        @assert ll[i′] != -1
```

Comment:
Make sure parent has already changed `-1` into something meaningful

Reply:
What about setting `ll[1] = 0.0` in `fwguidtree!`?

Reply:
I just make sure the field does not contain `-1` which would indicate a mistake or bad luck